src/isomorphic/classic/element/ReactElement.js

```diff
@@ -159,6 +160,37 @@ ReactElement.createElement = function(type, config, children) {
     }
   }
 
+  if (__DEV__) {
+    // Create dummy `key` and `ref` property to `props` to warn users
+    // against its use
+    Object.defineProperty(props, 'key', {
+      get: function () {
+        warning(
+          false,
+          '%s: `key` is not a prop. Trying to access it will result ' +
+          'in `undefined` being returned. If you need to access the same ' +
+          'value within the child component, you should pass it as a different ' +
+          'prop. (http://stackoverflow.com/questions/33682774/how-to-access-the' +
```

Comment:
We generally use fb.me urls that point to gists.  I created one for you to use: https://fb.me/react-special-props